src/common/common.h

```diff
@@ -152,6 +152,11 @@ UniqueID globally_unique_id(void);
 
 typedef UniqueID ObjectID;
 
+#ifdef __cplusplus
+size_t hashObjectID(const ObjectID &key);
+bool operator==(const ObjectID& x, const ObjectID& y);
```

Comment:
Wes, I don't think it will have an effect (with O3 optimizations turned on), but we'll test it out.